tests/RunApkTests.targets

```diff
@@ -20,6 +20,7 @@
   <Import Project="..\tests\CodeGen-Binding\Xamarin.Android.JcwGen-Tests\Xamarin.Android.JcwGen-Tests.projitems" Condition=" '$(AotAssemblies)' != 'True' " />
   <Import Project="..\tests\locales\Xamarin.Android.Locale-Tests\Xamarin.Android.Locale-Tests.projitems"  Condition=" '$(AotAssemblies)' != 'True' " />
   <Import Project="..\tests\Xamarin.Forms-Performance-Integration\Droid\Xamarin.Forms.Performance.Integration.Droid.projitems"  Condition=" '$(Configuration)' == 'Release' And '$(AotAssemblies)' != 'true' " />
+  <Import Project="..\tests\EmbeddedDSOs\EmbeddedDSO\EmbeddedDSO.projitems" Condition=" '$(AotAssemblies)' != 'True' " />
```

Comment:
This raises a good question: 

Will this work with AOT'd assemblies?  Meaning, does mono use us to find/load the appropriate `libaot-mscorlib.dll.so`/etc. files?  I *think* it does...but it would *probably* be a good idea to run these tests when `$(AotAssemblies)`=True, to *ensure* that it works.

I think the `Condition` should be removed.

Reply:
Yes, it does. See here for a debug output for this PR on Android O:

```
09-07 12:31:04.940 21101 21101 D Mono    : Image addref mscorlib[0xf205fcc0] -> mscorlib.dll[0xf2a8cd00]: 2
09-07 12:31:04.941 21101 21101 D Mono    : Prepared to set up assembly 'mscorlib' (mscorlib.dll)
09-07 12:31:04.941 21101 21101 D linker  : dlopen(name="mscorlib.dll.so", flags=0x3, extinfo=(null), caller="/data/app/com.xamarin.NotificationChannels-WCBF9YhPQcjyUVKBgGNbYA==/base.apk!/lib/x86/libmonosgen-2.0.so", caller_ns=classloader-namespace@0xf7579110) ...
09-07 12:31:04.941 21101 21101 D linker  : ... dlopen failed: library "mscorlib.dll.so" not found
09-07 12:31:04.941 21101 21101 D linker  : dlerror set to "dlopen failed: library "mscorlib.dll.so" not found"
09-07 12:31:04.941 21101 21101 D linker  : dlopen(name="/data/user/0/com.xamarin.NotificationChannels/files/.__override__/mscorlib.dll.so", flags=0x1, extinfo=(null), caller="/data/app/com.xamarin.NotificationChannels-WCBF9YhPQcjyUVKBgGNbYA==/base.apk!/lib/x86/libmonodroid.so", caller_ns=classloader-namespace@0xf7579110) ...
09-07 12:31:04.941 21101 21101 D linker  : ... dlopen failed: library "/data/user/0/com.xamarin.NotificationChannels/files/.__override__/mscorlib.dll.so" not found
09-07 12:31:04.941 21101 21101 D linker  : dlerror set to "dlopen failed: library "/data/user/0/com.xamarin.NotificationChannels/files/.__override__/mscorlib.dll.so" not found"
09-07 12:31:04.941 21101 21101 D linker  : dlopen(name="/data/user/0/com.xamarin.NotificationChannels/files/.__override__/mscorlib.dll.so", flags=0x1, extinfo=(null), caller="/data/app/com.xamarin.NotificationChannels-WCBF9YhPQcjyUVKBgGNbYA==/base.apk!/lib/x86/libmonodroid.so", caller_ns=classloader-namespace@0xf7579110) ...
09-07 12:31:04.941 21101 21101 D linker  : ... dlopen failed: library "/data/user/0/com.xamarin.NotificationChannels/files/.__override__/mscorlib.dll.so" not found
09-07 12:31:04.941 21101 21101 D linker  : dlerror set to "dlopen failed: library "/data/user/0/com.xamarin.NotificationChannels/files/.__override__/mscorlib.dll.so" not found"
09-07 12:31:04.941 21101 21101 D linker  : dlopen(name="/data/user/0/com.xamarin.NotificationChannels/files/.__override__/mscorlib.dll.so", flags=0x1, extinfo=(null), caller="/data/app/com.xamarin.NotificationChannels-WCBF9YhPQcjyUVKBgGNbYA==/base.apk!/lib/x86/libmonodroid.so", caller_ns=classloader-namespace@0xf7579110) ...
09-07 12:31:04.941 21101 21101 D linker  : ... dlopen failed: library "/data/user/0/com.xamarin.NotificationChannels/files/.__override__/mscorlib.dll.so" not found
09-07 12:31:04.941 21101 21101 D linker  : dlerror set to "dlopen failed: library "/data/user/0/com.xamarin.NotificationChannels/files/.__override__/mscorlib.dll.so" not found"
09-07 12:31:04.941 21101 21101 D linker  : dlopen(name="/data/app/com.xamarin.NotificationChannels-WCBF9YhPQcjyUVKBgGNbYA==/base.apk!/lib/x86/mscorlib.dll.so", flags=0x1, extinfo=(null), caller="/data/app/com.xamarin.NotificationChannels-WCBF9YhPQcjyUVKBgGNbYA==/base.apk!/lib/x86/libmonodroid.so", caller_ns=classloader-namespace@0xf7579110) ...
09-07 12:31:04.942 21101 21101 D linker  : ... dlopen failed: library "/data/app/com.xamarin.NotificationChannels-WCBF9YhPQcjyUVKBgGNbYA==/base.apk!/lib/x86/mscorlib.dll.so" not found
09-07 12:31:04.942 21101 21101 D linker  : dlerror set to "dlopen failed: library "/data/app/com.xamarin.NotificationChannels-WCBF9YhPQcjyUVKBgGNbYA==/base.apk!/lib/x86/mscorlib.dll.so" not found"
09-07 12:31:04.942 21101 21101 D linker  : dlopen(name="/data/user/0/com.xamarin.NotificationChannels/files/.__override__/libaot-mscorlib.dll.so", flags=0x1, extinfo=(null), caller="/data/app/com.xamarin.NotificationChannels-WCBF9YhPQcjyUVKBgGNbYA==/base.apk!/lib/x86/libmonodroid.so", caller_ns=classloader-namespace@0xf7579110) ...
09-07 12:31:04.942 21101 21101 D linker  : ... dlopen failed: library "/data/user/0/com.xamarin.NotificationChannels/files/.__override__/libaot-mscorlib.dll.so" not found
09-07 12:31:04.942 21101 21101 D linker  : dlerror set to "dlopen failed: library "/data/user/0/com.xamarin.NotificationChannels/files/.__override__/libaot-mscorlib.dll.so" not found"
09-07 12:31:04.942 21101 21101 D linker  : dlopen(name="/data/user/0/com.xamarin.NotificationChannels/files/.__override__/libaot-mscorlib.dll.so", flags=0x1, extinfo=(null), caller="/data/app/com.xamarin.NotificationChannels-WCBF9YhPQcjyUVKBgGNbYA==/base.apk!/lib/x86/libmonodroid.so", caller_ns=classloader-namespace@0xf7579110) ...
09-07 12:31:04.942 21101 21101 D linker  : ... dlopen failed: library "/data/user/0/com.xamarin.NotificationChannels/files/.__override__/libaot-mscorlib.dll.so" not found
09-07 12:31:04.942 21101 21101 D linker  : dlerror set to "dlopen failed: library "/data/user/0/com.xamarin.NotificationChannels/files/.__override__/libaot-mscorlib.dll.so" not found"
09-07 12:31:04.942 21101 21101 D linker  : dlopen(name="/data/user/0/com.xamarin.NotificationChannels/files/.__override__/libaot-mscorlib.dll.so", flags=0x1, extinfo=(null), caller="/data/app/com.xamarin.NotificationChannels-WCBF9YhPQcjyUVKBgGNbYA==/base.apk!/lib/x86/libmonodroid.so", caller_ns=classloader-namespace@0xf7579110) ...
09-07 12:31:04.942 21101 21101 D linker : ... dlopen failed: library "/data/user/0/com.xamarin.NotificationChannels/files/.__override__/libaot-mscorlib.dll.so" not found
```

Reply:
> > Will this work with AOT'd assemblies?
> 
> Yes, it does

Yay.

However, your snippet doesn't answer my question: for loading AOT'd assemblies, I'd expect a message that contains a substring along the lines of `base.apk!/lib/x86/libaot-mscorlib.dll.so`.  I do not see any message [in your snippet](
https://github.com/xamarin/xamarin-android/pull/2154/files#r217501222) which contains that substring.  Instead, every mention of `libaot-mscorlib.dll.so` is for the `.__override__` directory, *not* within the `.apk`.  As such, the snippet does *not* appear to answer my question/concern.

Additionally, you didn't reply to my final suggestion:

> I think the Condition should be removed.

Thoughts?

Reply:
@grendello wrote:

> Regarding the condtion - I don't know why it was put in there for all the other projects, so I can't make a decision 

The `Condition` is in there to not "unduly" prolong the build, because we believe that there isn't sufficient merit in running those tests under AOT scenarios (because AOT is time consuming).

Note that `Mono.Android-Tests.projitems` *doesn't* have the `Condition`, and thus *is* run for all configurations, because we believe there *is* merit in running that test under AOT.

I think that there *is* merit in running `EmbeddedDSO.projitems` when `$(AotAssemblies)`==True, because this test is very low-level and could easily break AOT if not done carefully.

As such, I think we should do:

```xml
<Import Project="..\tests\EmbeddedDSOs\EmbeddedDSO\EmbeddedDSO.projitems" />
```